event-based-simulators/main/event_based_simulators.py

```diff
@@ -419,20 +384,88 @@ def __send_event(self, event_fragment, timestamp = None):
             return newTimestamp
 
     def is_in_productionTime(self):
-        #if there are no shiftplans for a device, it should not be affected by production-time
+        # if there are no shiftplans for a device, it should not be affected by production-time
         has_no_shiftplan = True
         for shiftplan in shiftplans:
             if shiftplan.locationId == self.locationId:
                 has_no_shiftplan = False
                 for timeslot in shiftplan.recurringTimeSlots:
                     if timeslot.slotType == "PRODUCTION":
                         now = datetime.utcnow()
-                        start = datetime.strptime(timeslot.slotStart, shiftplan_dateformat)
-                        end = datetime.strptime(timeslot.slotEnd, shiftplan_dateformat)
+                        start = datetime.strptime(timeslot.slotStart, Shiftplan.dateformat)
+                        end = datetime.strptime(timeslot.slotEnd, Shiftplan.dateformat)
                         if start < now and end > now:
                             return True
         return has_no_shiftplan
 
+    # Measurements functions #
+    def measurement_functions(self, measurement_definition, task):
+        MachineSimulator.generate_measurement(self=self, measurement_definition=measurement_definition)
+        MachineSimulator.send_measurements(self=self, measurement_definition=measurement_definition)
+
+    def generate_measurement(self, measurement_definition):
+        log.info(f"Generating value of measurement {measurement_definition.get('series')} of device {self.model.get('id')}")
+        self.simulated_data = []
```

Comment:
why do we need to create an array if we only ever add a single element?

Reply:
I created this method when thinking about sending a collection of measurements but then I changed to send only a single one so it is still left like that. Since it doesn't affect the code until now so I forget about it

Reply:
> this simulated data array contains all the measurement definitions of a machine so it can have multiple values

are you sure? b/c I can see only one `.append()` invocation and a couple of accesses with `simulated_data[0]`

Reply:
After carefully checking, I realize what you found is correct. I will make modification about it also